File: BT Tool/markform old/markform/frm_sub_monitoring.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace markform
{
    public partial class frm_sub_monitoring : Form
    {
        CustomClass cf = new CustomClass();
        markDBOClass.SQLClass db = new markDBOClass.SQLClass();
        private DataGridView[] dgvarr;
        private string[] statArr;
        private string clientName = "";
        private string CurDt = "";
        public frm_sub_monitoring()
        {
            InitializeComponent();
            this.Tag = "monitoringlog";
            dgvarr = new DataGridView[] {dgv_btsched, dgv_prsched, dgv_ccsched, dgv_stsched, dgv_forbt,
            dgv_forpr, dgv_forcc, dgv_forst, dgv_forts, dgv_foraudit, dgv_ready, dgv_done};
            statArr = new string[] {"bt sched", "pr sched", "cc sched", "st sched", "for bt", "for pr", "for cc",
                "for st", "for ts","for audit", "ready", "done"};

        }
        public void SetSelectedTab(string clientname, int TabIndex, string dt)
        {
            this.tabctrl.SelectedIndex = TabIndex;
            this.CurDt = dt;
            clientName = clientname;
            this.Text = "Monitoring - " + clientname;
            LoadData();
            Notif();
        }
        private void LoadData()
        {
            this.Cursor = Cursors.AppStarting;
            db.SQLDependency(false);
            tss_status.Text = "Fetching data...";
            string curdt = CurDt, tssStr = "";//DateTime.Now.ToString("MM/dd/yyyy");
            DataTable dt = new DataTable();
            string qry = "SELECT branch,sFile,duration,status FROM dbo.Main WHERE client LIKE '" + clientName + "' AND LOWER([status]) NOT LIKE 'archive'";

            DataTable btscheddt = new DataTable(); DataTable prscheddt = new DataTable(); DataTable ccscheddt = new DataTable();
            DataTable stscheddt = new DataTable(); DataTable forbtdt = new DataTable(); DataTable forprdt = new DataTable();
            DataTable forccdt = new DataTable(); DataTable forstdt = new DataTable(); DataTable fortsdt = new DataTable();
            DataTable forauditdt = new DataTable(); DataTable readydt = new DataTable(); DataTable donedt = new DataTable();
            List<DGVData> dgvdata = new List<DGVData>()
            {
                new DGVData{dgv = dgv_btsched, dataTable = btscheddt},new DGVData{dgv = dgv_prsched, dataTable = prscheddt},
                new DGVData{dgv = dgv_ccsched, dataTable = ccscheddt},new DGVData{dgv = dgv_stsched, dataTable = stscheddt},
                new DGVData{dgv = dgv_forbt, dataTable = forbtdt},new DGVData{dgv = dgv_forpr, dataTable = forprdt},
                new DGVData{dgv = dgv_forcc, dataTable = forccdt},new DGVData{dgv = dgv_forst, dataTable = forstdt},
                new DGVData{dgv = dgv_forts, dataTable = fortsdt},new DGVData{dgv = dgv_foraudit, dataTable = forauditdt},
                new DGVData{dgv = dgv_ready, dataTable = readydt},new DGVData{dgv = dgv_done, dataTable = donedt}
            };

            foreach (var dgvd in dgvdata)
            {
                dgvd.dataTable.Columns.Add("Branch", typeof(string));
                dgvd.dataTable.Columns.Add("Filename", typeof(string));
                dgvd.dataTable.Columns.Add("Duration", typeof(string));
            }

            try
            {
                dt = db.query(qry);
                foreach(DataRow row in dt.Rows)
                {
                    object[] obj = row.ItemArray;
                    string stat = obj[3].ToString().ToLower();
                    dgvdata[Array.IndexOf(statArr, stat)].dataTable.Rows.Add(obj[0], obj[1], obj[2]);
                }

                foreach (var dgvd in dgvdata)
                {
                    dgvd.dgv.DataSource = dgvd.dataTable;
                    dgvd.dgv.ClearSelection();
                    cf.RowsNumber(dgvd.dgv);
                }
                tssStr = "Ready!";
            }
            catch (Exception ex)
            {
                tssStr = "Error Encountered!";
                cf.WriteToFile("{0}==>" + ex.ToString() + "\n\n" + qry, Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments) + "\\err.txt");
                MessageBox.Show(this, "Exception:\n" + ex.Message, "Error Encountered", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            tss_status.Text = tssStr;
            this.Cursor = Cursors.Default;
        }

        private void tabctrl_SelectedIndexChanged(object sender, EventArgs e)
        {
            DataGridView dgv = dgvarr[tabctrl.SelectedIndex];
            cf.RowsNumber(dgv);
        }


        void OnNewMessage()
        {
            ISynchronizeInvoke i = (ISynchronizeInvoke)this;
            if (i.InvokeRequired)
            {
                markDBOClass.SQLClass.NewMessage tempDelegate = new markDBOClass.SQLClass.NewMessage(OnNewMessage);
                i.BeginInvoke(tempDelegate, null);
                return;
            }
            db.OnNewMessage -= OnNewMessage;
            LoadData();
            Notif();
        }

        private void Notif()
        {
            string curdate = CurDt;//DateTime.Now.ToString("MM/dd/yyyy");
            db.SQLDependency(true);
            db.OnNewMessage += new markDBOClass.SQLClass.NewMessage(OnNewMessage);
            db.query("SELECT Id,status,dateServ FROM dbo.Main");
        }
    }
}

File: BT Tool/BT Tool/Forms/frm_sub_monitoring.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Windows.Forms

Public Class frm_sub_monitoring

#Region "Class"

    Private cf As New markform.CustomClass
    Private db As New markform.SQLClass

    Public Class DGVData
        Public Property dgv As DataGridView
        Public Property dataTable As DataTable
    End Class
#End Region

#Region "Variables"
    Private dgvArr As DataGridView()
    Private statArr As String()
    Private clientName As String = ""
    Private CurDt As String = ""
#End Region

#Region "Sub"


    Public Sub New()

        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        Me.Tag = "Monitoring Log"
        dgvArr = New DataGridView() {dgv_btsched, dgv_prsched, dgv_ccsched, dgv_stsched, dgv_forbt, dgv_forpr, dgv_forcc, dgv_forst, dgv_foraudit, dgv_ready, dgv_done}
        statArr = New String() {"bt sched", "pr sched", "cc shed", "st sched", "for bt", "for pr", "for cc", "for st", "for ts", "for audit", "ready", "done"}
    End Sub

    Public Sub SetSelectedTab(ByVal clientname As String, ByVal TabIndex As Integer, ByVal dt As String)
        Me.tabctrl.SelectedIndex = TabIndex
        Me.CurDt = dt
        clientname = clientname
        Me.Text = "Monitoring - " & clientname
        LoadData()
        Notif()
    End Sub

    Public Sub LoadData()

        Me.Cursor = Cursors.AppStarting
        db.SQLDependency(False)
        tss_status.Text = "Fetching Data..."
        Dim crdt As String = CurDt, tssStr As String = ""
        Dim dt As New DataTable()
        Dim qry As String = "SELECT branch,sFile,duration,status FROM dbo.Main WHERE client LIKE '" & clientName + "' AND LOWER([status]) NOT LIKE 'archive'"

        Dim btscheddt As New DataTable(), prscheddt As New DataTable(), ccscheddt As New DataTable(), stscheddt As New DataTable(), forbtdt As New DataTable(),
        forprdt As New DataTable(), forccdt As New DataTable(), forstdt As New DataTable(), fortsdt As New DataTable(), forauditdt As New DataTable(),
        readydt As New DataTable(), donedt As New DataTable()

        Dim dgvdata As New List(Of frm_file_eval.DGVData)()
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_btsched, .dataTable = btscheddt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_btsched, .dataTable = prscheddt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_ccsched, .dataTable = ccscheddt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_stsched, .dataTable = stscheddt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_forbt, .dataTable = forbtdt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_forpr, .dataTable = forprdt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_forcc, .dataTable = forccdt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_forst, .dataTable = forstdt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_forts, .dataTable = fortsdt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_foraudit, .dataTable = forauditdt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_ready, .dataTable = readydt})
        dgvdata.Add(New frm_file_eval.DGVData() With {.dgv = dgv_done, .dataTable = donedt})

        For Each dgvd In dgvdata
            dgvd.dataTable.Columns.Add("Branch", GetType(String))
            dgvd.dataTable.Columns.Add("Filename", GetType(String))
            dgvd.dataTable.Columns.Add("Duration", GetType(String))
        Next

        Try
            dt = db.query(qry)
            For Each row As DataRow In dt.Rows
                Dim obj As Object() = row.ItemArray
                Dim stat As String = obj(3).ToString.ToLower
                dgvdata(Array.IndexOf(statArr, stat)).dataTable.Rows.Add(obj(0), obj(1), obj(2))
            Next

            For Each dgvd In dgvdata
                dgvd.dgv.DataSource = dgvd.dataTable
                dgvd.dgv.ClearSelection()
                cf.RowsNumber(dgvd.dgv)
            Next
            tssStr = "Ready!"

        Catch ex As Exception
            tssStr = "Error Encountered!"
            cf.WriteToFile("{0}==>" & ex.ToString() & "\n\n" & qry, Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments) & "\\err.txt")
            MessageBox.Show(Me, "Exception:\n" + ex.Message, "Error Encountered", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try

    End Sub


    Private Sub OnNewMessage()
        Dim i As ISynchronizeInvoke = CType(Me, ISynchronizeInvoke)

        If i.InvokeRequired Then
            Dim tempDelegate As markform.SQLClass.NewMessage = New markform.SQLClass.NewMessage(AddressOf OnNewMessage)
            i.BeginInvoke(tempDelegate, Nothing)
            Return
        End If

        AddHandler db.OnNewMessage, AddressOf OnNewMessage
        LoadData()
        Notif()
    End Sub

    Private Sub Notif()
        Dim curdate As String = CurDt
        db.SQLDependency(True)
        AddHandler db.OnNewMessage, AddressOf OnNewMessage
        db.query("SELECT Id,status,dateServ FROM dbo.Main")
    End Sub

#End Region

#Region "Events"



#End Region

#Region "Functions"

#End Region


End Class